### PR TITLE
Add user/pass authentication support for Druid

### PR DESCRIPTION
**Pull request by Jakdaw**
## What type of PR is this? (check all applicable)

- [X] Feature

## Description

Add support for user/password authentication when connecting to Druid.

### CODE REVIEW

File: redash/query_runner/druid.py [hunk 1]
```diff
@@ -48,7 +55,9 @@ def run_query(self, query, user):
         connection = connect(host=self.configuration['host'],
                              port=self.configuration['port'],
                              path='/druid/v2/sql/',
-                             scheme=self.configuration['scheme'])
+                             scheme=self.configuration['scheme'],
+                             user=self.configuration['user'],
```
Comment (arikfr) [inline redash/query_runner/druid.py]:
```suggestion
                             user=self.configuration.get('user'),
```

`user` might be missing from the config dict (and same for password)

Also, do you know if the client library has no side effects if passing `None` to `user`? What about an empty string?

File: redash/query_runner/druid.py [hunk 2]
```diff
@@ -48,7 +55,9 @@ def run_query(self, query, user):
...
+                             password=self.configuration['password'])
```
Comment (arikfr) [inline redash/query_runner/druid.py]:
```suggestion
                             password=self.configuration.get('password'))
```

File: redash/query_runner/druid.py [hunk 3]
```diff
@@ -48,7 +55,9 @@ def run_query(self, query, user):
         connection = connect(host=self.configuration['host'],
                              port=self.configuration['port'],
                              path='/druid/v2/sql/',
-                             scheme=self.configuration['scheme'])
+                             scheme=self.configuration['scheme'],
```
Comment (arikfr) [inline redash/query_runner/druid.py]:
```suggestion
                             scheme=self.configuration.get('scheme', 'http'),
```

(this was an issue before your PR, but worth addressing)

**Comment by Jakdaw** [comment]
Library treats None like no parameter being passed... empty string it doesn't.... so I guess I need to go look at how self.configuration is populated.

**Comment by Jakdaw** [comment]
That's neat!

**Comment by arikfr** [comment]
Thanks!

**Comment by dlevison** [comment]
To overcome this issue in redash v8.0.0 you can add the username and password in the host field:
<user>:<pass>@druid-host
Not great security wise but works.

**Comment by arikfr** [comment]
@dlevison clever :)

**Pull request merged**